streampark-console/streampark-console-service/src/main/resources/mapper/core/ApplicationMapper.xml

```diff
@@ -133,6 +133,14 @@
              limit 1
     </select>
 
+    <select id="getJobByClusterId" resultType="java.lang.Integer" parameterType="java.lang.Long">
+        SELECT
+            count(1)
+        FROM t_flink_app
+        WHERE flink_cluster_id = #{clusterId}
+            limit 1
+    </select>
+
```

Comment:
I have carefully considered here and it is indeed necessary to filter the status.

I want to filter the job status that is not `add` or `cancelled`. I need to explain why it is necessary to filter out tasks that are not `add` or `cancelled`:

Because during the execution of this SQL statement, jobs in other states can be considered running or preparing to run in the `flink cluster`, but it may be due to the issue of two scheduling threads being out of sync,unable to update the job status in a timely manner, it may not be possible to determine the affected jobs based on a certain status.

What do you think of `getAffectedJobsByClusterId` regarding select id.

Reply:
That sounds good~

Reply:
> * Do we need to filter the status of the job here ?
> 
> I sorted out briefly the background and logic. Assuming there is a job with a status of cancelled but current cluster information, will this job be counted if the cluster is abnormal?
> 
> * We'd better to get a better select-id name here.
> 
> Please correct me if I'm wrong.

Hi @RocMarshal @wolfboys , Regarding this issue, I found in the implementation that due to the asynchronous monitoring of the `application` and `flink cluster` threads, it is not possible to directly determine whether it is an affected job based on the state in the application. I may be looking for a more suitable implementation method. I plan to maintain the original logic regarding this PR. If it is a job deployed in the `flink cluster`, it will be defined as an affected job regardless of its status.

I look forward to your suggestions and responses very much.